### ISSUE TITLE
Bug: 2.0.1 unable to apply cspell on JSONC comments

### ISSUE THREAD

**Issue by lkraav** [eslint/json#275]
### Environment

ESLint version: 9.39.4
@eslint/json version: 2.0.1
Node version: 24.19
npm version: 11.19
Operating System: Linux

### Which language are you using?

jsonc

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
  {
    files: ['.devcontainer/devcontainer.json'],
    language: 'json/jsonc',
    ...json.configs.recommended,
    ...cspellRecommended,
    rules: {
      '@cspell/spellchecker': 2,
    },
  },
```
</details>

```js
<!-- Paste your code here -->
{
  // Features to add to the dev container. More info: https://containers.dev/features.
  "features": {
    "ghcr.io/devcontainers/features/copilot-cli:1": {},
    "ghcr.io/devcontainers/features/github-cli:1": {
      // GH queries intermittenly demand unexpected authentication, try APT.
      "installDirectlyFromGitHubRelease": false
    }
  },
}
```

### What did you expect to happen?

`cspell` should flag "intermittenly" typo.

### What actually happened?

eslint passes.

### Link to Minimal Reproducible Example

-

### Participation

- [ ] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

LLM hypothesis on cause: "@eslint/json/Momoa represents JSONC comments as LineComment tokens under Document.tokens; CSpell plugin skips tokens during AST traversal and only handles Line/Block comment nodes. checkScope cannot reach these tokens, so JSON comments are unsupported by this parser/plugin combination."

**Comment by DMartens** [comment]
JSON is a different [language](https://eslint.org/docs/latest/extend/languages) and as such rules written for JavaScript are not expected to work.
For example the core / JavaScript rules `sort-keys` or `key-spacing` also do not work when using the JSON language.
The CSpell plugin can be updated to support other languages by adding the needed listeners as you mentioned.
As such closing this issue as this is expected behavior.
